### ISSUE TITLE
QP vs QMCPACK energies

### ISSUE THREAD

**Issue by aannabe** [QMCPACK/qmcpack#1199]
Hello,
I've been seeing some inconsistencies with Quantum Package CIPSI energy vs QMCPACK VMC energy with the same wavefunction obtained via convert4qmc.
QP:
E =   -123.154750264452

QMCPACK:
                                LocalEnergy               Variance           ratio 
Fe.dump  series 0  -123.147392 +/- 0.001221   9.900219 +/- 0.130159   0.0804

It seems like this discrepancy is observed when naturals orbitals are used in QP instead of HF orbitals. I made sure that in both cases the number of determinants is the same.
As a side note, I had issues with the HDF5 file generated too. When used it gave:
`Fatal Error. Aborting at LCAOrbitalBuilder::putFromH5 Missing /KPTS_0/eigenset_0 from HDF5 File.`
Any suggestions to resolve these issues would be greatly helpful.

Files:
[Fe.zip](https://github.com/QMCPACK/qmcpack/files/2637682/Fe.zip)

**Comment by prckent** [comment]
Thanks for reporting this. Which version of QMCPACK are you running? Are you able to get agreement in the non-natural orbital case (HF or DFT orbitals)? This should definitely work and there should be excellent agreement between the energies. e.g. We have reproduced the Toulouse group water molecule results. Presumably in your case there is an issue or set of issues at the conversion step for natural orbitals. 

Note that additionally there are hints that we have problems with certain diffuse functions (#1145) in the QMC - @anbenali is working to understand and fix this currently.

**Comment by aannabe** [comment]
I am running develop branch (#1188). From a limited number of cases I've tried, HF orbitals energies agree.

**Comment by prckent** [comment]
@anbenali , @aannabe : Just reviewing this issue. Is it still a problem?

**Comment by aannabe** [comment]
When I was looking at these calculations, I sorted of convinced myself that QP converter was the source of issues here. Here is why. Above calculations' orbitals were obtained from Gaussian, which by default uses a spherical expansion of basis set. Then I tried GAMESS which uses cartesian expansion (although it can remove contaminations, I think it still preserves cartesian format), which had no issues like above. Finally, I tried Gaussian with cartesian which also didn't show any problems. Therefore, I concluded spherical basis set conversion to QP was the source of problems.
